server/app/services/mongo_pubsub.rb

```diff
@@ -21,14 +21,29 @@ def terminate
       stop
     end
 
+    # @return [Celluloid::Future]
+    def start
```

Comment:
I think `start` / `wait` is a bit confusing as an api. Method `wait` is called  from `process_subscription` and that actually triggers `process`.  If `process` should be protected always then maybe it should return a future?

Reply:
Refactored to remove the `@future ||= ...` lazy-init... no reason to not spawn the future right from `initialize`.